components/script/script_thread.rs

```diff
@@ -192,6 +192,43 @@ unsafe_no_jsmanaged_fields!(TaskQueue<MainThreadScriptMsg>);
 
 type NodeIdSet = HashSet<String>;
 
+/// A simple guard structure that restore the user interacting state when dropped
+#[derive(Default)]
+pub(crate) struct ScriptUserInteractingGuard<'a> {
+    was_interacting: bool,
+    script_thread: Option<&'a ScriptThread>,
```

Comment:
If we make `ScriptThread::is_user_interacting` a `Rc<Cell<bool>>`, we don't need to store a pointer to the ScriptThread. This allows us to remove all of the new `with_script_thread` calls that were added, which will make the changes much cleaner.